CodeEditModules/Modules/WorkspaceClient/src/Model/FileItem.swift

```diff
@@ -257,7 +257,12 @@ public extension WorkspaceClient {
             var fileUrl = self.url
             while FileItem.fileManger.fileExists(atPath: fileUrl.path) {
                 let previousName = fileUrl.lastPathComponent
-                fileUrl = fileUrl.deletingLastPathComponent().appendingPathComponent("\(previousName) copy")
+                let previousNameWithoutExtension = (previousName as NSString).deletingPathExtension
+                let filextension = fileUrl.pathExtension
+                let fileName = filextension.isEmpty ?
+                previousName : previousNameWithoutExtension
+                fileUrl = fileUrl.deletingLastPathComponent()
+                fileUrl.appendPathComponent("\(fileName)_copy.\(filextension)")
```

Comment:
if `fileextension` is empty then this would produce an additional `.`

Reply:
better use appendPathExtension

Reply:
Done